### ISSUE TITLE
Epoch 0 exited with error: signal: segmentation fault (core dumped)

### ISSUE THREAD

**Issue by rnkhouse** [istio/istio#32065]
(**NOTE**: This is used to report product bugs:
  To report a security vulnerability, please visit <https://istio.io/about/security-vulnerabilities>
  To ask questions about how to use Istio, please visit <https://discuss.istio.io>)

**Bug description**
I am getting this error in one of my applications sidecar proxy.
>Epoch 0 exited with error: signal: segmentation fault (core dumped)
The app is breaking after this error:

2021-04-08T22:05:00.635788Z	critical	envoy backtrace	Caught Segmentation fault, suspect faulting address 0x0
2021-04-08T22:05:00.635829Z	critical	envoy backtrace	Backtrace (use tools/stack_decode.py to get line numbers):
2021-04-08T22:05:00.635834Z	critical	envoy backtrace	Envoy version: dc78069b10cc94fa07bb974b7101dd1b42e2e7bf/1.15.1-dev/Clean/RELEASE/BoringSSL
2021-04-08T22:05:00.637013Z	critical	envoy backtrace	#0: __restore_rt [0x7f331c5998a0]
2021-04-08T22:05:00.672872Z	critical	envoy backtrace	#1: Envoy::Network::UdpListenerImpl::send() [0x56463bfdac15]
2021-04-08T22:05:00.761018Z	critical	envoy backtrace	#2: Envoy::Extensions::UdpFilters::DnsFilter::DnsFilter::sendDnsResponse() [0x56463bcffe81]
2021-04-08T22:05:00.772073Z	critical	envoy backtrace	#3: std::__1::__function::__func<>::operator()() [0x56463bd03e58]
2021-04-08T22:05:00.863711Z	critical	envoy backtrace	#4: Envoy::Extensions::UdpFilters::DnsFilter::DnsFilterResolver::onResolveTimeout() [0x56463bd059a2]
2021-04-08T22:05:00.874557Z	critical	envoy backtrace	#5: event_process_active_single_queue [0x56463c434737]
2021-04-08T22:05:00.966298Z	critical	envoy backtrace	#6: event_base_loop [0x56463c4332ae]
2021-04-08T22:05:01.060877Z	critical	envoy backtrace	#7: Envoy::Server::WorkerImpl::threadRoutine() [0x56463bfbdf84]
2021-04-08T22:05:01.071843Z	critical	envoy backtrace	#8: Envoy::Thread::ThreadImplPosix::ThreadImplPosix()::{lambda()#1}::__invoke() [0x56463c4e4573]
2021-04-08T22:05:01.071914Z	critical	envoy backtrace	#9: start_thread [0x7f331c58e6db]
2021-04-08T22:05:01.194612Z	info	sds	resource:default connection is terminated: rpc error: code = Canceled desc = context canceled
2021-04-08T22:05:01.194649Z	error	sds	Remote side closed connection
2021-04-08T22:05:01.194613Z	info	sds	resource:ROOTCA connection is terminated: rpc error: code = Canceled desc = context canceled
2021-04-08T22:05:01.194675Z	error	sds	Remote side closed connection
2021-04-08T22:05:01.195620Z	error	Epoch 0 exited with error: signal: segmentation fault (core dumped)
2021-04-08T22:05:01.195644Z	info	No more active epochs, terminating

[ ] Docs
[ ] Installation
[x] Networking
[ ] Performance and Scalability
[ ] Extensions and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure
[ ] Upgrade

**Expected behavior**
It should run the application with sidecar proxy without errors.

**Steps to reproduce the bug**
Install istio 1.7.3 version, create application with sidecar in it. Generate some load in the app and you will see above error.

**Version** (include the output of `istioctl version --remote` and `kubectl version --short` and `helm version --short` if you used Helm)
1.7.3
I also tried with 1.7.8

**How was Istio installed?**
istioctl with generate manifect

**Environment where the bug was observed (cloud vendor, OS, etc)**
AKS

**Comment by bianpengyuan** [comment]
Do you have some EnvoyFilter sets up DNS filter?

**Comment by rnkhouse** [comment]
@bianpengyuan  No.

**Comment by bianpengyuan** [comment]
Does it crash loop or crash intermittently? Would it be possible if you can get a config dump from the proxy? https://github.com/istio/istio/wiki/Troubleshooting-Istio#collecting-information-2

Istio does not send any DnsFilter config AFAIK, which is what causes this crash. Is there anything special with your Istio set up?

**Comment by howardjohn** [comment]
We sent the dns filter for 1.7 I think but it was off by default. So this should only happen with an experimental feature, and it's fixed in all supported Istio versions so I don't think there isn't anything to do here.

**Comment by bianpengyuan** [comment]
> We sent the dns filter for 1.7 I think but it was off by default. So this should only happen with an experimental feature, and it's fixed in all supported Istio versions so I don't think there isn't anything to do here.

I see. sorry did not realize 1.7 does dns capture that way.

**Comment by howardjohn** [comment]
Yeah we switched since it crashes all the time :-)

**Comment by rnkhouse** [comment]
I Readiness and Liveness probe errors in describe pod:

Warning  Unhealthy  6m37s (x1479 over 9h)  kubelet  Liveness probe failed: Get http://10.2.0.250:15020/app-health/myapp/livez: net/http: request canceled (Client.Timeout exceeded while awaiting headers)
  Warning  Unhealthy  104s (x1484 over 10h)  kubelet  Readiness probe failed: Get http://10.2.0.250:15020/app-health/myapp/readyz: net/http: request canceled (Client.Timeout exceeded while awaiting headers)

I didn't define above path for readiness and liveness probes.

@bianpengyuan  There is no extra setup. We use external cms endpoint in this app which is already mentioned in ServiceEntry our outboundTrafficPolicy is REGISTRY_ONLY.

**Comment by ronakthamor** [comment]
@howardjohn  This is how I installed istio:

Config file:
# https://istio.io/docs/reference/config/istio.operator.v1alpha1/#IstioOperatorSpec
---
apiVersion: install.istio.io/v1alpha1
kind: IstioOperator
spec:
  profile: default
  components:
    pilot:
      k8s:
        replicaCount: 1
        resources:
          requests:
            cpu: 100m # override from default 500m
            memory: 512Mi # ... default 2048Mi
    egressGateways:
    - name: istio-egressgateway
      enabled: true
      k8s:
        replicaCount: 1
        resources:
          requests:
            cpu: 500m
            memory: 128Mi
    ingressGateways:
    - name: istio-ingressgateway
      enabled: true
      k8s:
        replicaCount: 1
        resources:
          requests:
            cpu: 500m
            memory: 128Mi
        overlays:
        - kind: Service
          name: istio-ingressgateway
          patches:
          - path: spec.loadBalancerIP
            value: LB_PUBLIC_IP
          - path: spec.externalTrafficPolicy
            value: Local
        serviceAnnotations:
          service.beta.kubernetes.io/azure-load-balancer-resource-group: LB_RG_NAME
    - name: istio-internal-ingressgateway
      enabled: true
      #label:
      #  istio: istio-internal-ingressgateway
      k8s:
        replicaCount: 1
        resources:
          requests:
            cpu: 50m
            memory: 128Mi
        overlays:
        - kind: HorizontalPodAutoscaler
          name: istio-internal-ingressgateway
          patches:
          - path: metadata.labels.app
            value: istio-internal-ingressgateway
          - path: metadata.labels.istio
            value: internal-ingressgateway
          - path: spec.scaleTargetRef.name
            value: istio-internal-ingressgateway
        - kind: Deployment
          name: istio-internal-ingressgateway
          patches:
          - path: metadata.labels.app
            value: istio-internal-ingressgateway
          - path: metadata.labels.istio
            value: internal-ingressgateway
          - path: spec.selector.matchLabels.app
            value: istio-internal-ingressgateway
          - path: spec.selector.matchLabels.istio
            value: internal-ingressgateway
          - path: spec.template.metadata.labels.app
            value: istio-internal-ingressgateway
          - path: spec.template.metadata.labels.istio
            value: internal-ingressgateway
        - kind: Service
          name: istio-internal-ingressgateway
          patches:
          - path: metadata.labels.app
            value: istio-internal-ingressgateway
          - path: metadata.labels.istio
            value: internal-ingressgateway
          - path: spec.selector.app
            value: istio-internal-ingressgateway
          - path: spec.selector.istio
            value: internal-ingressgateway
        #- kind: Service
        #  name: istio-internal-ingressgateway
        #  patches:
        #  - path: spec.externalTrafficPolicy
        #    value: Local
        serviceAnnotations:
          service.beta.kubernetes.io/azure-load-balancer-internal: "true"
  meshConfig:
    enableAutoMtls: true
    accessLogFile: "/dev/stdout"
    accessLogEncoding: JSON
    outboundTrafficPolicy:
      mode: REGISTRY_ONLY
    #enablePrometheusMerge: false # https://istio.io/latest/docs/ops/integrations/prometheus/
    policyCheckFailOpen: false
    enableTracing: true
    defaultConfig:
      # NOTE: https://istio.io/latest/news/releases/1.8.x/announcing-1.8/upgrade-notes/#istio-coredns-plugin-deprecation
      proxyMetadata:
        ISTIO_META_DNS_CAPTURE: "true"
        ISTIO_META_PROXY_XDS_VIA_AGENT: "true"
    #  tracing:
    #    datadog:
    #      #address: "datadog-cluster-agent.default.svc.cluster.local:5005"
    #      address: "$(HOST_IP):8126"
    #    zipkin:
    #      address: "zipkin:9411"
      sds:
        enabled: true
  # Ref: https://istio.io/v1.5/docs/reference/config/installation-options/#grafana-options
  values:
    global:
      logging:
        level: "default:info"
      controlPlaneSecurityEnabled: true
      proxy:
        logLevel: warning
        resources:
          requests:
            cpu: 100m
            memory: 256Mi
          limits:
            cpu: 200m
            memory: 384Mi
      #privileged: true
    gateways:
      istio-ingressgateway:
        sds:
          enabled: true
    # tracing is deprecated and will be removed in version 1.8; use the samples/addons/ deployments instead
    #tracing:
    #  enabled: true
    #  provider: zipkin
    #  zipkin:
    #    resources:
    #      requests:
    #        cpu: 50m
    #        memory: 256Mi
    #      limits:
    #        cpu: 100m
    #        memory: 512Mi

***

    ./bin/istioctl manifest generate -f ../config.yaml > generated_manifest.yaml
    kubectl apply -f generated_manifest.yaml
    kubectl apply -f samples/addons -n istio-system

**Comment by howardjohn** [comment]
```
ISTIO_META_DNS_CAPTURE: "true"
ISTIO_META_PROXY_XDS_VIA_AGENT: "true"
```

This does it. Please upgrade or remove that, this is a known issue on an EOL version (https://istio.io/latest/news/support/announcing-1.7-eol-final/) in an experimental feature, so while its a valid bug we won't fix it in 1.7. Thanks!
